src/ObservationDims.jl

```diff
@@ -1,5 +1,159 @@
 module ObservationDims
 
-greet() = print("Hello World!")
+using NamedDims
+using AxisArrays
+using Distributions
+
+export obs_arrangement, organise_obs
+export SingleObs, IteratorOfObs, ArraySlicesofObs
+export MatrixRowsOfObs, MatrixColsOfObs
+
+"""
+    type ObsArrangement
+
+Defines the orientation of data that is expected by a function
+"""
+abstract type ObsArrangement end
+
+struct SingleObs <: ObsArrangement end
+struct IteratorOfObs <: ObsArrangement end
+struct ArraySlicesOfObs{D} <: ObsArrangement end
+
+const MatrixRowsOfObs = ArraySlicesOfObs{1}
+const MatrixColsOfObs = ArraySlicesOfObs{2}
+
+
+"""
+    organise_obs(f, data; obsdim=nothing)
+
+Organise the `data` according to the `ObsArrangement` expected by some function `f`.
+"""
+function organise_obs(f, data; obsdim=nothing)
+    return organise_obs(obs_arrangement(f), data; obsdim=obsdim)
```

Comment:
obs_arrangement is not defined anywhere

Reply:
also we should probably aim for 100% test coverage, since that should be easy :)